src/main/java/org/terasology/metalrenegades/ai/system/TimeSensitiveSystem.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.metalrenegades.ai.system;
+
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.metalrenegades.ai.component.TimeSensitiveComponent;
+import org.terasology.registry.In;
+import org.terasology.world.time.WorldTimeEvent;
+
+/**
+ * Provides current world time information to entities with {@link TimeSensitiveComponent}.
+ */
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class TimeSensitiveSystem extends BaseComponentSystem {
+
+    /**
+     * Time of in-game sunrise
+     */
+    private static final float NIGHT_END = 0.250f;
+
+    /**
+     * Time of in-game sunset
+     */
+    private static final float NIGHT_BEGIN = 0.720f;
+
+    @In
+    private EntityManager entityManager;
+
+    @ReceiveEvent
+    public void onWorldTimeEvent(WorldTimeEvent worldTimeEvent, EntityRef entityRef) {
+        for (EntityRef entity : entityManager.getEntitiesWith(TimeSensitiveComponent.class)) {
+            TimeSensitiveComponent timeSensitiveComponent = entity.getComponent(TimeSensitiveComponent.class);
```

Comment:
This loop is called many, many times and therefore resource intensive. I think it would be better to determine "isNight" when it's actually needed. The world/day time can be retrieved at any point via injection, so the `TimeSensititveComponent/System` is probably not required.